rust/gitxetcore/src/xetmnt/mod.rs

```diff
@@ -0,0 +1,488 @@
+pub mod xetfs_bare;
+
+#[cfg(unix)]
+pub mod xetfs_write;
+
+use crate::config::XetConfig;
+use crate::errors::{GitXetRepoError, Result};
+use cas::output_bytes;
+use nfsserve::tcp::*;
+use prometheus;
+use prometheus_dict_encoder::DictEncoder;
+use std::path::Path;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use tokio::process::Command;
+use tokio::sync::mpsc;
+use tokio::time;
+
+use tracing::{error, info};
+
+#[cfg(unix)]
+use tracing::warn;
+
+pub fn check_for_mount_program() -> bool {
+    if cfg!(target_os = "macos") {
+        // we can always mount on mac
+        true
+    } else if cfg!(target_os = "linux") {
+        // we use /sbin/mount.nfs on Linux
+        // TODO: the command here is only for Ubuntu. Might need to check
+        // for the other OSes
+        if !Path::new("/sbin/mount.nfs").exists() {
+            error!("Unable to locate /sbin/mount.nfs");
+            error!("Ubuntu: Install the nfs client package with 'apt install nfs-common'");
+            error!("Redhat/Fedora: Install the nfs client package with 'yum install nfs-utils'");
+            false
+        } else {
+            true
+        }
+    } else if cfg!(target_os = "windows") {
+        // TODO: must change this to deteck if the NFS Client is installed.  I don't see how to do this
```

Comment:
Maybe DISM would be work?

Run follwing command 

```DISM /online /get-features /format:table /english |findstr /I "nfs"```

and get result:
```
ServicesForNFS-ClientOnly                   | Disabled
ClientForNFS-Infrastructure                 | Disabled
NFS-Administration                          | Disabled
```
Problem is DISM need run as  admin.

Reply:
Thanks @willser, opened #149 with this suggestion.